lib/core/src/Cardano/Pool/DB/Model.hs

```diff
@@ -219,7 +219,7 @@ mReadSystemSeed db@PoolDatabase{seed} =
 
 mReadCursor :: Int -> ModelPoolOp [BlockHeader]
 mReadCursor k db@PoolDatabase{pools} =
-    let allHeaders = foldMap snd $ Map.toList pools
+    let allHeaders = fold pools
```

Comment:
Using the fact that Map is Foldable and the list (values in the pools map) is Monoid we can avoid unnecessary traversal.

Reply:
I'd expect the compiler to optimize list fusions for me but maybe that's too naive / too strong faith in GHC :grimacing: